service/host_helpers.go

```diff
@@ -0,0 +1,56 @@
+package service
+
+import (
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/evergreen/model/user"
+	"github.com/evergreen-ci/evergreen/units"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/pkg/errors"
+)
+
+const (
+	InvalidStatusError             = "'%v' is not a valid status"
+	DecommissionStaticHostError    = "Cannot decommission static hosts"
+	HostTerminationQueueingError   = "Error starting background job for host termination"
+	HostUpdateError                = "Error updating host"
+	HostTerminationQueueingSuccess = "Host %v successfully queued for termination"
+	HostStatusUpdateSuccess        = "Host status successfully updated from '%v' to '%v'"
+	HostStatusWriteConfirm         = "Successfully updated host status"
+	UnrecognizedAction             = "Unrecognized action: %v"
+)
+
+func modifyHostStatus(h *host.Host, opts *uiParams, u *user.DBUser) (string, error) {
+	env := evergreen.GetEnvironment()
+	queue := env.RemoteQueue()
+
+	switch opts.Action {
+	case "updateStatus":
+		currentStatus := h.Status
+		newStatus := opts.Status
+		if !util.StringSliceContains(validUpdateToStatuses, newStatus) {
+			return "", errors.Errorf(InvalidStatusError, newStatus)
```

Comment:
By returning errors like this, you've changed the Status codes that get returned. I'd recommend having this function return rest.APIError, with the status code set properly, and have the caller return the correct status code/message using APIError's fields